### ISSUE TITLE
clean up the `help` a bit

### ISSUE THREAD

**Issue by joyously** [jj-vcs/jj#3156]
## Description
There are some inconsistencies in the `help` output.
1. The `version` command is so vague, and since this is "version control", it seems like it should not be showing the global options. I think it should self-reference the executable so it doesn't look like it's a version of the repo.
![version-Screenshot_2024-02-27_20-16-26](https://github.com/martinvonz/jj/assets/7957680/7e506712-48fe-4365-a704-7c95b9f5873c)

2.  There are 4 uses of `repo`, (chmod, git, init, status), but this isn't helpful for those that need to use a translator to get the whole idea.  
3.  `next` and `prev` end with a period; the rest do not.
4.  `undo` uses back ticks and refers to jj, which it probably should not.
5. Although the glossary says that revision is a synonym for commit, commit is both a verb and noun, and in `jj` the command is an alias. The noun commit is used **8 times** in the command help, while revision is used **16 times**. 
![commit-Screenshot_2024-02-27_20-20-02](https://github.com/martinvonz/jj/assets/7957680/106ae929-985c-40d5-9455-637b49bf6bbe)
![revision-Screenshot_2024-02-27_20-19-34](https://github.com/martinvonz/jj/assets/7957680/f6e45624-8358-4ea7-b221-84e18aee4db2)
_The white part is 80 columns._
The glossary page has a table of contents, but it has commit **72 times** and revision 6 times. Considering that "revset" (also not a word that will translate) is baked in, maybe revision should be the real thing and commit is the synonym.
6.  I wonder if alphabetical listing of commands is the most useful. `help` is at the end though. It could be sorted by type of object affected or workflow (help and init at top) or basic CRUD order.

_I just noticed that the contributing.md refers to `checkout`, which was removed, so this might be in other places also._
## Specifications

- Version: 0.14.0

**Comment by Jollywatt** [comment]
I think being consistent with the language of "revisions" instead of "commits" would be a key point for learners. I got quite hung up on that until I found the glossary!

"Revision": a set of changes; jj's equivalent of git'a commit

"Commit" (verb): `jj describe` then `jj new`

"Commit" (noun): a git commit, not a jj revision

(I'm in love with jj by the way — I found it two days ago and I already know I'll never go back to git!)

**Comment by martinvonz** [comment]
I actually think of "revision" and "commit" (noun) as synonyms.

> "Revision": a set of changes; jj's equivalent of git's "commit"

And here it sounds like you also do.

> "Commit" (noun): a git commit, not a jj revision

But here it sounds like you don't. What nuance am I missing?

> I think being consistent with the language of "revisions" instead of "commits" would be a key point for learners. I got quite hung up on that until I found the glossary!

Do you mean that it's confusing that we use both "revisions" and "commits" for the same thing? (Maybe the answer will be obvious based on you answer to my question above.)

**Comment by Jollywatt** [comment]
> Do you mean that it's confusing that we use both "revisions" and "commits" for the same thing?

At least to me, it was confusing — or at least, not as beautifully simple as it could be. All the words one tends to come across in version control ("commit", "move", "rebase", "revert", "reset", "split" "blame", …) tend to have _very_ specific meanings, so I assumed that the word "revision" was carefully picked to denote something distinct from a "commit". This was reinforced when I learned that jj’s "to commit" was an alias for "describe then make new", so named to cater to git-knowers.

> But here it sounds like you don't. What nuance am I missing?

Ah, I was meaning perhaps we should reserve the word "commit" (as a noun) for when talking about git’s commits. I’m sure jj’s revision and git’s commits are technically isomorphic, but they still “feel” different to use, and it helped my mental model to keep them separate.

This is all subjective, but bottom line, I think reducing synonymous jargon terms is good. Not necessarily across the whole docs, but at least for the one-line command descriptions.

**Comment by joyously** [comment]
Maybe the nuance is that in the English dictionary, commit is only a verb. The VCS world has made it into a noun. 
While `jj` currently uses a Git backend, the help text should be in `jj` terms, and not in the current backend's terms. So the alias `commit` is good for those used to Git, aligning revset with revision would be best.

**Comment by joyously** [comment]
> This is my mental model after two days of knowing about jj

Rereading @Jollywatt 's reply, maybe he didn't get the right model since he seems to be saying that "revisions equals change" but also that "commits and revisions don't line up".

Jujutsu documentation says that the focus is on tracking "changes", which is good, and the decision _not_ to use the ambiguous word "change" for a command or in the help text is **very good** (unlike Pijul-- that word is confusing). 
I think the subtlety of mixing implementation details in command output and explanations makes it harder to convey the "`jj` way" of doing things.

**Comment by Jollywatt** [comment]
> maybe he didn't get the right model

Wait, what's not right about that model? o.o

**Comment by joyously** [comment]
> Wait, what's not right about that model?

It's so subtle you can't see it from what I said? For `jj`, you used "revisions are made", which is correct but that also means "commits are made" since they are synonyms, and "there can be multiple revisions per `jj` change". 
If there were no Git backend, but a `jj` native backend, it would just be "changes".

**Comment by Jollywatt** [comment]
I think we are in fact on the same page. I'm just advocating for not having "revisions" and "commits" as synonyms. My reason is that, if they were synonymous, we'd have this:

|  | Checkpoints in history | When editing a file
|----|------|---
| jj | changes, change IDs | _commits_ are made
| git | _commits_ | index becomes dirty |

...which is a little confusing, because now "commit" refers to different concepts in the jj versus git mental model.

**Comment by joyously** [comment]
I don't think it's quite the same, since I'm advocating for keeping implementation details like the words "revisions" and "commit" out of the explanations entirely. But I do favor "revision" because it's a real noun that goes with "revision set" (revset) and "change", if one has to be mentioned.

**Comment by necauqua** [comment]
In my head revisions are things identified by change ids, and commits are things identified by commit ids.

Commits are immutable objects in the git content addressable store, every time we _rewrite_ a revision/commit we actually create a new commit and hide the old one and the revision gets mutated.

We can list all commits that the revision has been with `obslog`.
Every time a snapshot happens a new commit is created, the working copy revision is amended.

Basically git has _revisions_ - but they are only in your head, the commits you amend change yet they represent the same work, for which things like gerrit and jj added the change-id thing.

**Comment by joyously** [comment]
> In my head revisions are things identified by change ids, and commits are things identified by commit ids.

That all sounds good, but doesn't match the glossary saying that they are synonyms.

**Comment by Jollywatt** [comment]
> In my head revisions are things identified by change ids, and commits are things identified by commit ids.

This is why I think we should standardise. It's too easy to think "revisions" and "commits" are different.
